tool/tsh/common/db.go

```diff
@@ -71,22 +72,25 @@ func onListDatabases(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
-	var proxy *client.ProxyClient
+	var clusterClient *client.ClusterClient
 	err = client.RetryWithRelogin(cf.Context, tc, func() error {
-		proxy, err = tc.ConnectToProxy(cf.Context)
+		clusterClient, err = tc.ConnectToCluster(cf.Context)
```

Comment:
Interesting pattern.

I wonder if this is an attempt to avoid excessive retries, or if we should provide an API to just connect to the cluster with the retries and return the final client.

Reply:
Oh these are just musings, I don't expect changes related to this comment here.